apps/clearance_ui/src/components/PurlDetails.tsx

```diff
@@ -18,23 +18,25 @@ type Props = {
 
 const PurlDetails = ({ purl }: Props) => {
     const parsedPurl = parsePurlAndQualifiers(purl);
-    const packageName =
+    const mainPurl =
+        "pkg:" +
+        (parsedPurl.type ? parsedPurl.type + "/" : "") +
```

Comment:
Per PURL spec, there can never be a PURL without a type. If the type is unknown, it [should be named `generic`](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#generic). With that in mind, we should be able to omit the conditional here.

Reply:
Btw, I think you can also utilize the packageurl-js package here as well [https://www.npmjs.com/package/packageurl-js#constructing](https://www.npmjs.com/package/packageurl-js#constructing) to construct the clean purl by setting the subpath and qualifiers to undefined